src/main/java/org/gridsuite/ds/server/service/DynamicSimulationWorkerService.java

```diff
@@ -115,12 +123,31 @@ public void updateResult(UUID resultUuid, DynamicSimulationResult result) {
                 DynamicSimulationStatus.CONVERGED :
                 DynamicSimulationStatus.DIVERGED;
 
-        resultService.updateResult(resultUuid, timeSeries, timeLineSeries, status);
+        resultService.updateResult(resultUuid, timeSeries, timeLineSeries, status, outputState);
     }
 
     @Override
     protected void saveResult(Network network, AbstractResultContext<DynamicSimulationRunContext> resultContext, DynamicSimulationResult result) {
-        updateResult(resultContext.getResultUuid(), result);
+        // read dump file
+        byte[] outputState = null;
+        Path dumpDir = Optional.ofNullable(resultContext.getRunContext().getDynamicSimulationParameters())
+                .map(parameters -> parameters.getExtension(DynawoSimulationParameters.class))
+                .map(dynawoSimulationParameters -> ((DynawoSimulationParameters) dynawoSimulationParameters).getDumpFileParameters().dumpFileFolder())
+                .orElse(null);
+        if (dumpDir != null) {
+            try (Stream<Path> files = Files.list(dumpDir)) {
+                Path dumpFile = files.findFirst().orElse(null);
+                if (dumpFile != null) {
+                    // ZIP output state
+                    outputState = Utils.zip(dumpFile);
+                }
+
+            } catch (IOException e) {
+                throw new DynamicSimulationException(DUMP_FILE_ERROR, "Error while reading dump file");
+            }
+        }
+
+        updateResult(resultContext.getResultUuid(), result, outputState);
```

Comment:
move and extract this logic from saveResult into small helper methods to reduce the complexity of the new code
1) Path dumpDir = getDumpFilePath(resultContext);
2) byte[] outputState = null; and  if (dumpDir != null) { outputState = zipDumpFile(dumpDir); }
3) updateResult(resultContext.getResultUuid(), result, outputState);

Reply:
DONE in https://github.com/gridsuite/dynamic-simulation-server/pull/116/commits/85a0baaed9feae2f53cfbfbf62f944076e27df09